packages/flutter/lib/src/material/theme_data.dart

```diff
@@ -470,9 +470,69 @@ class ThemeData extends Diagnosticable {
        assert(bottomSheetTheme != null),
        assert(popupMenuTheme != null);
 
-  // Warning: make sure these properties are in the exact same order as in
-  // hashValues() and in the raw constructor and in the order of fields in
-  // the class and in the lerp() method.
+  /// Create a [ThemeData] based on the colors in the given [colorScheme].
+  ///
+  /// It will derive appropriate values for all the [ThemeData] properties
+  /// based on [colorScheme] and [textTheme].
+  ///
+  /// [colorScheme] can not be [null].
+  ///
+  /// If [colorScheme.brightness] is [Brightness.dark] then
+  /// [ThemeData.applyElevationOverlayColor] will be set to [true] to support
+  /// the Material dark theme method for indicating elevation by overlaying
+  /// a semi-transparent white color on top of the surface color.
+  ///
+  /// This is the recommended method to theme your application. As we move
+  /// forward we will be converting all the widget implementations to only use
+  /// colors or colors derived from those in [ColorScheme].
+  ///
+  /// {@tool sample}
+  /// This example will set up an application to use the baseline Material
+  /// Design light and dark themes.
+  ///
+  /// ```dart
+  /// MaterialApp(
+  ///   theme: ThemeData.from(colorScheme: ColorScheme.light()),
+  ///   darkTheme: ThemeData.from(colorScheme: ColorScheme.dark()),
+  /// )
+  /// ```
+  /// {@end-tool}
+  ///
+  /// See <https://material.io/design/color/> for
+  /// more discussion on how to pick the right colors.
+  factory ThemeData.from({
+    @required ColorScheme colorScheme,
+    TextTheme textTheme,
+  }) {
+    assert(colorScheme != null);
+
+    final bool isDark = colorScheme.brightness == Brightness.dark;
+
+    // For surfaces that use primary color in light themes and surface color in dark
+    final Color primarySurfaceColor = isDark ? colorScheme.surface : colorScheme.primary;
+    final Color onPrimarySurfaceColor = isDark ? colorScheme.onSurface : colorScheme.onPrimary;
+
+    return ThemeData(
+      brightness: colorScheme.brightness,
+      primaryColor: primarySurfaceColor,
+      primaryColorBrightness: ThemeData.estimateBrightnessForColor(primarySurfaceColor),
+      primaryColorLight: colorScheme.primaryVariant,
+      primaryColorDark: colorScheme.primaryVariant,
```

Comment:
This might be a good place to add a comment about why it's OK to set both of these values to the same thing, i.e. where we use primaryColorLight,Dark, etc.